crypto/ml_kem/ml_kem_ipd_ref_common/indcpa.c

```diff
@@ -217,7 +217,11 @@ void indcpa_keypair_derand(ml_kem_params *params,
   uint8_t nonce = 0;
   polyvec a[KYBER_K_MAX], e, pkpv, skpv;
 
-  hash_g(buf, coins, KYBER_SYMBYTES);
+  uint8_t coins_with_domain_separator[KYBER_SYMBYTES + 1];
+  memcpy(coins_with_domain_separator, coins, KYBER_SYMBYTES);
+  coins_with_domain_separator[KYBER_SYMBYTES] = params->k;
```

Comment:
footnote 1  in [FIPS 203](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.203.pdf#subsection.5.1)